File: python-js/intro-to-code/retention-assignments/algo-anagrams-i/python/character_match.py
# Don't forget to run your tests!

def is_character_match(string1, string2):
    string1 = string1.replace(" ", "").upper()
    string2 = string2.replace(" ", "").upper()
    
    if len(string1) != len(string2):
        return False
    else:
        for letter in string1:
            if letter not in string2:
                return False
            idx = string2.index(letter)
            string2 = string2[:idx] + string2[idx+1:]
        return True

File: python-js/intro-to-code/retention-assignments/algo-anagrams-i/js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {
    let modified_string1 = string1.replaceAll(" ", "").toUpperCase();
    let modified_string2 = string2.replaceAll(" ", "").toUpperCase();
    if (modified_string1.length != modified_string2.length) {
        return false;
    }
    for (let c of modified_string1) {
        let idx = modified_string2.indexOf(c);
        if (idx < 0) {
            return false;
        }
        modified_string2 = modified_string2.replace(c, "");
    }
    return true;
};
